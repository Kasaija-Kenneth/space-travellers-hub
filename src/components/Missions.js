import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { fetchMission, joinMission } from '../redux/missions/missions';
import './Mission.modules.css';

const Missions = () => {
  const missions = useSelector((state) => state.missions.missionList);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMission());
  }, [dispatch]);

  const handleClick = ({ target }) => {
    const { type, id } = target.dataset;
    const payload = { type, id };
    dispatch(joinMission(payload));
  };

  return (
    <div>
      <table>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr className="mission-container" key={mission.mission_id}>
              <td className="name-field">{mission.mission_name}</td>
              <td className="description-field">{mission.description}</td>
              <td className="status-field">
                <p className={mission.join ? 'active' : ''}>
                  {mission.join ? 'Active Member' : 'NOT A MEMBER'}
                </p>
              </td>
              <td className="action-container">
                <button
                  type="button"
                  data-id={mission.mission_id}
                  data-type={mission.join ? 'leave' : 'join'}
                  className={mission.join ? 'btn active' : 'btn'}
                  onClick={handleClick}
                >
                  {mission.join ? 'Leave Mission' : 'Join Mission'}
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Missions;
