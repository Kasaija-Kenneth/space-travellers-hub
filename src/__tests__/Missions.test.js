import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Missions from '../components/Missions';
import store from '../redux/configureStore';

describe('snapshots testing', () => {
  test('snapshots for Missions component', () => {
    const rendererComponents = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponents).toMatchSnapshot();
  });
});
