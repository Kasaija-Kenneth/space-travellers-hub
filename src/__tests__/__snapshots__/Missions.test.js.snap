// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots testing snapshots for Missions component 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
